From: Christian Weisgerber <naddy@mips.inka.de>
Date: Mon, 4 Jul 2022 15:13:11 +0000 (+0000)
Subject: fix potential type mismatch between format specifier and argument
X-Git-Url: https://git.gameoftrees.org/gitweb/?p=got.git;a=commitdiff_plain;h=b9a36ae5

fix potential type mismatch between format specifier and argument

Cast printf argument of type time_t to long long to match the %lld
format specifier on platforms where this might not be the case.
---

diff --git a/lib/sigs.c b/lib/sigs.c
index 693a3a27..ac1b2637 100644
--- a/lib/sigs.c
+++ b/lib/sigs.c
@@ -204,7 +204,7 @@ got_tag_write_signed_data(BUF *buf, struct got_tag_object *tag,
 	got_date_format_gmtoff(gmtoff, sizeof(gmtoff),
 	    got_object_tag_get_tagger_gmtoff(tag));
 	if (asprintf(&tagger, "%s %lld %s", got_object_tag_get_tagger(tag),
-	    got_object_tag_get_tagger_time(tag), gmtoff) == -1) {
+	    (long long)got_object_tag_get_tagger_time(tag), gmtoff) == -1) {
 		err = got_error_from_errno("asprintf");
 		goto done;
 	}
